### PR TITLE
fix(bookmarks): align restored bookmarks with Android KJVA ordinals

### PR DESCRIPTION
**Pull request by primetheus**
## Summary

Fixes Android bookmark restore semantics by separating Android's source-module `book` value from iOS's display book name. Restored and native Bible bookmarks now render, highlight, navigate, and export from KJVA ordinals instead of relying on the repurposed display `book` field.

## Scope

- `BibleBookmark` persistence now stores source module initials separately in `bookInitials`.
- Remote sync, Android database backup restore, patch replay, and snapshot export preserve Android's module-initials-or-NULL `BibleBookmark.book` contract.
- Bookmark list rows, reader highlights, My Notes, accessibility state, and bridge payloads resolve references from KJVA ordinal columns.
- Native bookmark creation stores source ordinals plus Android-compatible KJVA storage ordinals.
- Tests cover restore normalization, outbound export, KJVA querying, bridge payloads, My Notes projection, reset cleanup, and divergent-canon clamping.

## Contract References

- Android stores `BibleBookmark.book` as `AbstractPassageBook.initials` or NULL, not as a Bible book display name.
- Android renders bookmark references from versification ordinals, with KJVA as the shared cross-module storage domain.
- Android treats NULL-source-book bridge payloads as whole-verse KJVA payloads with no text offsets.

## Change Details

- Added `BibleBookmark.bookInitials` for Android source module identity while keeping `BibleBookmark.book` display-facing for legacy iOS fallback paths.
- Removed the display-book predicate from bookmark overlap queries; KJVA ordinals are globally unique and match Android's lookup model.
- Changed list display, navigation targets, reader chapter queries, and My Notes payloads to use KJVA ordinals first.
- Changed outbound Android snapshots and patch comparison to export source initials or preserved NULL, never display names such as `Genesis`.
- Added nearest-addressable KJVA clamping when exact source-to-KJVA projection fails, preventing write-side corruption for divergent versification edge cases.

## Validation Evidence

- `xcodebuild test -scheme BibleCoreTests` passed: 292 tests, 0 failures.
- `xcodebuild test -scheme BibleUITests -only-testing:BibleUITests/BookmarkReaderBridgeTests` passed: 33 tests, 0 failures.
- `python3 scripts/check_repo_standards.py docblocks --all-files` passed.
- `git diff --check` passed.
- KJVA constants were recomputed against libsword `canon.h` / `canon_kjva.h`.

## Risks / Follow-Ups

- iOS still lacks a full JSword-style `toV11n(KJVA)` mapping engine. This PR prevents wrong-domain ordinal corruption and fixes the mainstream #356 restore path, but divergent canons such as LXX, Vulgate, and Synodal still need a dedicated mapping engine and legacy-row backfill follow-up.
- `bookInitials` is a new SwiftData/CloudKit-synced attribute with an empty-string default, so existing rows migrate as unknown-source bookmarks until recreated or restored.

## Issue Closure

Closes #356

### CODE REVIEW

File: Sources/BibleUI/Sources/BibleUI/Bible/BibleReaderController.swift [hunk 1]
```diff
@@ -4063,13 +4137,13 @@ public final class BibleReaderController: NSObject, BibleBridgeDelegate {
      Opens the chapter-level My Notes document in the current pane.
      */
     public func bridge(_ bridge: BibleBridge, openMyNotes v11n: String, ordinal: Int) {
-        loadMyNotesDocument(jumpToOrdinal: ordinal)
+        loadMyNotesDocument(jumpToOrdinal: kjvaMyNotesOrdinal(v11nName: v11n, sourceOrdinal: ordinal) ?? ordinal)
     }
```
[the remark this answers is not in the document]
Comment (primetheus) [reply Sources/BibleUI/Sources/BibleUI/Bible/BibleReaderController.swift:4155]: Fixed in 8210c17f. `openMyNotes` now passes only the KJVA projection result into `loadMyNotesDocument`, so a failed source-to-KJVA conversion opens My Notes without a jump target instead of reusing a source-domain ordinal. Covered by `testReaderOpenMyNotesOmitsJumpTargetWhenSourceOrdinalCannotConvertToKJVA`; full PR CI is green.

File: Sources/BibleUI/Sources/BibleUI/Bible/BibleReaderBookmarkActionCoordinator.swift [hunk 2]
```diff
@@ -179,17 +200,34 @@ struct BibleReaderBookmarkActionCoordinator {
...
+        let sourceStart = min(startOrdinal, effectiveEndOrdinal)
+        let sourceEnd = max(startOrdinal, effectiveEndOrdinal)
+        let storageRange = kjvaOrdinalRange(sourceStart, sourceEnd) ?? (start: sourceStart, end: sourceEnd)
         let bookmark = bookmarkService.addParagraphBreakBibleBookmark(
             bookInitials: bookInitials,
             startOrdinal: startOrdinal,
             endOrdinal: endOrdinal,
+            kjvOrdinalStart: storageRange.start,
+            kjvOrdinalEnd: storageRange.end,
+            v11n: currentV11n(),
```
[the remark this answers is not in the document]
Comment (primetheus) [reply Sources/BibleUI/Sources/BibleUI/Bible/BibleReaderBookmarkActionCoordinator.swift]: Fixed in 8210c17f. Paragraph-break Bible bookmark creation now uses the same normalized `sourceStart`/`sourceEnd` persistence contract before writing KJVA fields, preventing zero-ended or reversed source ranges from being stored. Covered by `testBookmarkActionCoordinatorPersistsNormalizedSourceRangeForParagraphBreakBibleBookmark`; full PR CI is green.
